pkg/ssh/connection_golang.go

```diff
@@ -272,23 +273,54 @@ func ValidateAndConfigure(uri *url.URL, iden string) (*ssh.ClientConfig, error)
 	if err != nil {
 		return nil, err
 	}
-	keyFilePath := filepath.Join(homedir.Get(), ".ssh", "known_hosts")
-	known, err := knownhosts.New(keyFilePath)
-	if err != nil {
-		return nil, fmt.Errorf("creating host key callback function for %s: %w", keyFilePath, err)
-	}
 
+	var keyErr *knownhosts.KeyError
 	cfg := &ssh.ClientConfig{
-		User:            uri.User.Username(),
-		Auth:            authMethods,
-		HostKeyCallback: known,
-		Timeout:         tick,
+		User: uri.User.Username(),
+		Auth: authMethods,
+		HostKeyCallback: ssh.HostKeyCallback(func(host string, remote net.Addr, pubKey ssh.PublicKey) error {
+			keyFilePath := filepath.Join(homedir.Get(), ".ssh", "known_hosts")
+			known, err := knownhosts.New(keyFilePath)
+			if err != nil {
+				if os.IsNotExist(err) && insecureIsMachineConnection {
```

Comment:
@mtrmac @cdoern correct me if I am wrong, but I think the intention behind `insecureIsMachineConnection`, is not to require host key validation. I think it would be simpler to either check right away and return nil before loading hosts, or provide an alternate machineHostKeyCallback handler which is effectively a no-op. The way it seems to work now is that keys aren't added if the file doesn't exist, but they are added if it does. This would lead to surprising behavior differences. Like I can recreate the machine if the known_host file does not exist, but if it does exist it will add and I will get a host validation error after recreating (rm / init) a machine.

Reply:
yes you are correct in that assumption, however, we also want to enable the ability for machines to be validated. I think @mtrmac and I discussed that we only want machine to "be excused" from validation for a few podman versions while people get adjusted, but only if we hit an error. This can be changed though if we just wan to remove them from the equation by default

Reply:
@n1hility I’m not sure we have this mapped out, can you help us with the details of what happens on machine rm/init? Does that create a new machine _with the same host name_? (And is that a somewhat frequent operation or an extremely rare corner case? I guess closer to the former, without any dat.)

If we need to account for re-created machines, maybe we, in the ideal state, really shouldn’t add entries to the global known_hosts — and use a private per-machine known_hosts that is deleted along with the machine.

Reply:
Sorry for the delay, was traveling abroad and in meetings. The way machine works is the connection entries all use localhost, but with a randomly generated port. The failure scenarios would be random.

Your approach would be ideal, a separate known_hosts file. 

Recreating is fairly common, it’s an easy way to upgrade or start with a clean slate. 

Since you are trying to get this out the door fast, my recommendation would be to adjust your PR to just skip everything with known hosts with podman machine. That would avoid the regression from 4.2 (even if intermittent and under specific scenarios)

Reply:
@n1hility so if machine... use `ssh.InsecureIgnoreHostKey()`? @mtrmac do you think that makes sense?

personally I think a few random failures (I have not run into any yet with testing) is a good first step in comparison to accomplishing nothing. We were using `InsecureIgnoreHostKey` for a while here

Reply:
> @n1hility so if machine... use `ssh.InsecureIgnoreHostKey()`?

Isn’t that _basically_ what we have now in this PR, if `known_hosts` does not exist at all? Apart from a warning, that is.

In that sense, we are already OK with doing that.

Yes,
```go
if insecureIsMachineConnection {
  callback = ssh.InsecureIgnoreHostKey
} else {
  callback = func() { /* enforce match in known_hosts, or add a new entry if there is none (or if the file is missing completely) */ }
}
```
seems like a good step forward for 4.3.

Reply:
@cdoern sure that would be fine. 

It’s true the odds are someone doesn’t run into this, but if they do it won’t be clear to the average podman machine user why it’s happening and what action to take to resolve it. To them they create a machine (perhaps not even via the CLI, perhaps via podman desktop) that succeeds. Afterwords thier podman command fails because of some ssh host key verification issue. They won’t necessarily know why ssh is even involved , since the interface is the podman command. That mysterious error will lead to a “podman machine is broken” bug reports.

In the case where they don’t have a known hosts they will get a scary warn message (may also lead to a bug report - “what does this mean?”).

In the machine case, there is not much to balance this out on the positive side. Host key verification doesn’t do much for you on locahost, and this iteration isn’t intentionally doing it across the board anyway. So IMO negatives are out saying the benefits of doing some incomplete checking.

Reply:
ok, I will push the version with ignoring key validation while on podman machine